tests/conftest.py

```diff
@@ -93,20 +105,31 @@ def lower_privs():
             os.setgid(1000)
             os.setuid(1000)
 
-        popen = subprocess.Popen(command_line, preexec_fn=lower_privs, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        popen = subprocess.Popen(
+            command_line, preexec_fn=lower_privs, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd="/tmp"
+        )
         try:
             # wait 2 seconds to ensure it starts
             popen.wait(2)
         except subprocess.TimeoutExpired:
             pass
         else:
+            _print_process_output(popen)
             raise Exception(f"Command {command_line} exited unexpectedly with {popen.returncode}")
 
         yield popen
+
+        # ensure, again, that it still alive (if it exited prematurely it might provide bad data for the tests)
+        try:
+            popen.wait(0)
+        except subprocess.TimeoutExpired:
+            pass
+        else:
+            _print_process_output(popen)
+            raise Exception(f"Command {command_line} exited unexpectedly during the test with {popen.returncode}")
+
```

Comment:
There are no tests that actively kill the process?

Reply:
Nope, why would we (it's the process being profiled, and these tests check just the profiling of it)